bpf/kmesh/probes/performance_probe.h

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Kmesh */
+
+#ifndef __KMESH_BPF_PERFORMANCE_MONITOR_H__
+#define __KMESH_BPF_PERFORMANCE_MONITOR_H__
+
+#include "bpf_common.h"
+
+enum {
+    SOCK_TRAFFIC_CONTROL = 1,
+    XDP_SHUTDOWN = 2,
+    ENABLE_ENCODING_METADATA = 3,
+};
+
+struct operation_usage_data {
+    __u64 start_time;
+    __u64 end_time;
+    __u32 operation_type;
+};
+
+struct operation_usage_key {
+    __u32 tid;
+    __u32 operation_type;
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, struct operation_usage_key);
+    __type(value, struct operation_usage_data);
+    __uint(max_entries, 1024);
+} kmesh_perf_map SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_RINGBUF);
+    __uint(max_entries, RINGBUF_SIZE);
+} kmesh_perf_info SEC(".maps");
+
+static inline void performance_report(struct operation_usage_data *data)
+{
+    struct operation_usage_data *info = NULL;
+    info = bpf_ringbuf_reserve(&kmesh_perf_info, sizeof(struct operation_usage_data), 0);
+    if (!info) {
+        BPF_LOG(ERR, PROBE, "bpf_ringbuf_reserve map proformance info failed\n");
+        return;
+    }
+    info->start_time = data->start_time;
+    info->end_time = data->end_time;
+    info->operation_type = data->operation_type;
+    bpf_ringbuf_submit(info, 0);
```

Comment:
you only report thes info to userspace, donot we care about which pid is operating on this?

Reply:
The _pid_tgid_ has been added to the labels.